python-checks/src/main/java/org/sonar/python/checks/regex/UnusedGroupNamesCheck.java

```diff
@@ -117,31 +118,51 @@ private void checkGroupAccesses(RegexParseResult regexParseResult, KnownGroupsCo
   }
 
   private void checkGroupNameOrNumberExists(RegexParseResult regexParseResult, KnownGroupsCollector groupsCollector, Expression argumentExpression) {
+    //Check if group accessed by name exists
     Optional.of(argumentExpression)
       .filter(StringLiteral.class::isInstance)
       .map(StringLiteral.class::cast)
       .map(StringLiteral::trimmedQuotesValue)
       .filter(Predicate.not(groupsCollector.byName::containsKey))
-      .map(NotExistedGroupUsagesCheck::getMessage)
-      .or(() -> Optional.of(argumentExpression)
-        .filter(NumericLiteral.class::isInstance)
-        .map(NumericLiteral.class::cast)
-        .map(NumericLiteral::valueAsLong)
-        .filter(Predicate.not(groupsCollector.byNumber::containsKey))
-        .map(NotExistedGroupUsagesCheck::getMessage))
-      .ifPresent(message -> {
-        IssueLocation issueLocation = PythonRegexIssueLocation.preciseLocation(regexParseResult.getResult(), message);
-        regexContext.addIssue(argumentExpression, message)
-          .secondary(issueLocation);
+      .ifPresent(notExistedGroupName -> {
+        var message = getGroupNameNotExistsMessage(notExistedGroupName);
+        var issue = regexContext.addIssue(argumentExpression, message);
+
+        if (groupsCollector.byName.isEmpty()) {
+          var secondaryLocation = PythonRegexIssueLocation.preciseLocation(regexParseResult.getResult(), NO_GROUP_NAMES_SECONDARY_MESSAGE);
+          issue.secondary(secondaryLocation);
+        } else {
+          groupsCollector.byName.forEach((groupName, group) -> {
+            var secondaryMessage = String.format(GROUP_NAME_SECONDARY_MESSAGE_FORMAT, groupName);
+            var secondaryLocation = PythonRegexIssueLocation.preciseLocation(group, secondaryMessage);
+            issue.secondary(secondaryLocation);
+          });
+        }
+      });
+
+    //Check if group accessed by number exists
```

Comment:
This should say `Check if group access by number doesn't have name`

Reply:
done